Makefile

```diff
@@ -104,15 +94,8 @@ kubefed2: $(BIN_DIR)
 controller-native: $(BIN_DIR)
 	$(BUILD_CONTROLLER_NATIVE)
 
-push:
-	$(DOCKER) push $(REGISTRY)/$(TARGET):$(GIT_VERSION)
-	if git describe --tags --exact-match >/dev/null 2>&1; \
-	then \
-		$(DOCKER) tag $(REGISTRY)/$(TARGET):$(GIT_VERSION) $(REGISTRY)/$(TARGET):$(GIT_TAG); \
-		$(DOCKER) push $(REGISTRY)/$(TARGET):$(GIT_TAG); \
-		$(DOCKER) tag $(REGISTRY)/$(TARGET):$(GIT_VERSION) $(REGISTRY)/$(TARGET):latest; \
-		$(DOCKER) push $(REGISTRY)/$(TARGET):latest; \
-	fi
+push: container
+	bash ./scripts/imagebuild.sh push
```

Comment:
I think it will be useful to be able to give your own image name [ IMAGE=xxxx ]. Most of the people invoking this will not push to the default (official) image location.

Reply:
@irfanurrehman, it is possible, like below:
```
$> REGISTRY=quay.io REPO=<user-account> make push
```

Reply:
Probably I wasn't clear the first time. I meant a canonical name which can be documented also and `IMAGE=<repo>/<username>/<image-name>:<tag>` as one override option seems to fit that better. Say the user now wants to override tag and actual image name not just `REGISTRY` and `REPO`, it wont be possible. This was missing from earlier Makefile too and its probably better to have it.

Reply:
Agree with @irfanurrehman. Coincidentally, I just went through implementing a script that changed repo and registry, and then needed to change the image name.